### PR TITLE
amber g auth

### PR DESCRIPTION
**Pull request by drujensen**
This change provides the ability to generate an authenticated resource.

Unfortunately, there is some templating that is duplicated with minor modifications.  We may want to look at a way to sharing templates between generators. 

To test:
```
amber new project --deps
cd project
amber g auth User name:string
crystal db/seeds.cr
```

This will generate a User model with scaffolding.
It forces generation of `email:string` and `encrypted_password:string` fields.
It adds the `SessionController` with login/create/logout methods with routes.
It adds a new `middleware/Authenticate` to the pipeline.
It adds a `db/seeds.cr` that will create a new user.
It adds a new field type `password` for correct generation of the forms.

### CODE REVIEW

**Comment by drujensen** [comment]
I"m not happy with the code duplication.  I am going to see if I can rework this so we are not duplicating the scaffold templates.

File: src/templates/app/src/{{name}}.cr [hunk 1]
```diff
@@ -2,7 +2,7 @@ require "amber"
 require "../config/*"
 require "./models/**"
 require "./mailers/**"
-require "./controllers/application_controller"
```
Comment (gabrielrios) [inline src/templates/app/src/{{name}}.cr]: I think this like is needed as per  Amber-Crystal/amber_cmd#55

Comment (drujensen) [inline src/templates/app/src/{{name}}.cr]: @gabrielrios Thanks for catching this.

**Comment by drujensen** [comment]
moved to amber

**Pull request closed without merging**